### PR TITLE
Level Door require level to walkthrough

### PR DESCRIPTION
**Pull request by PrinterLUA**
I find this way more promising, since players who are worhty to pass. Then they should just walk through and enter the door, instead of making walkstack of only max 1 player.

This should fix this issue:
https://github.com/otland/forgottenserver/issues/1591

### CODE REVIEW

**Comment by PrinterLUA** [comment]
Before i merge the commits, what do you think @marksamman

**Comment by PrinterLUA** [comment]
Before i merge the commits, what do you think @marksamman

**Comment by brunominervino** [comment]
Good idea @PrinterLUA, I will test.

File: data/movements/movements.xml [hunk 1]
```diff
@@ -35,95 +35,185 @@
 
 	<!-- (Level & quest) doors -->
 	<movevent event="StepOut" itemid="1224" script="closingdoor.lua" />
+	<movevent event="StepIn" itemid="1224" script="opendoor.lua" />
```
Comment (marksamman) [inline data/movements/movements.xml]: What's opendoor.lua?

Comment (PrinterLUA) [inline data/movements/movements.xml]: my fault, it was named opendoor, but then i changed to enterdoor.

Ill fix it tomorrow. Anything else you want me to change?

Comment (marksamman) [inline data/movements/movements.xml]: Yes. Split level doors and quest doors into two different script files, then you don't need the isInArray check.

**Comment by conde2** [comment]
Thanks Printer, great ideia !!!

**Comment by PrinterLUA** [comment]
@marksamman, done :)

**Comment by PrinterLUA** [comment]
@marksamman will it get merged, or shall we wait for the onStepIn/onStepOut enchantments?

**Comment by marksamman** [comment]
It doesn't need to wait for onStepIn/onStepOut changes, but you need to squash your commits. Have you also tested the changes?

**Comment by PrinterLUA** [comment]
@marksamman yes and works perfect.
preview: http://2.1m.yt/THWGfKD.gif

**Comment by wgrriffel** [comment]
@PrinterLUA It work as expected, only one thing i'd like to suggest to this PR, isn't good make a statement to gms and adms can walk through? like
`if player:getGroup():getAccess() == true then return false end`
imo it's should be onUse also, but it's just a thought, what you think?

**Pull request merged**